pydrive2/fs/spec.py

```diff
@@ -261,8 +261,8 @@ def _ids_cache(self):
     def _cache_path_id(self, path, *item_ids, cache=None):
         cache = cache or self._ids_cache
         for item_id in item_ids:
-            cache["dirs"][path].append(item_id)
             cache["ids"][item_id] = path
+            cache["dirs"][path].append(item_id)
```

Comment:
C: This is better to do in this order since we always refer first to `dirs` to decide id item is cached or not. Still looks fragile, and we still ideally should be using proper locking. I'll looking into this as a follow up.

Reply:
IIRC, this is not true in `find` where we use `ids` first.

Reply:
Ahh, I see that you also changed `find` below.